tests/unit/dependency-injection/schema-templates-pass-test.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Yoast\WP\SEO\Tests\Dependency_Injection;
+
+use Mockery;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Yoast\WP\SEO\Dependency_Injection\Schema_Templates_Pass;
+use Yoast\WP\SEO\Integrations\Schema_Blocks;
+use Yoast\WP\SEO\Tests\Unit\TestCase;
+
+/**
+ * Class Schema_Templates_Pass_Test.
+ *
+ * @group dependency-injection
+ *
+ * @coversDefaultClass \Yoast\WP\SEO\Dependency_Injection\Schema_Templates_Pass
+ */
+class Schema_Templates_Pass_Test extends TestCase {
+
+	/**
+	 * Represents the container builder.
+	 *
+	 * @var Mockery\MockInterface|ContainerBuilder
+	 */
+	protected $container_builder;
+
+	/**
+	 * Represents the instance to test.
+	 *
+	 * @var Schema_Templates_Pass
+	 */
+	protected $instance;
+
+	/**
+	 * Sets the instances to test.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->container_builder = Mockery::mock( ContainerBuilder::class );
+		$this->instance = new Schema_Templates_Pass();
+	}
+
+	/**
+	 * Tests the processing of the templates.
+	 *
+	 * @covers ::process
+	 * @covers ::get_templates
+	 */
+	public function test_process() {
+		$schema_blocks_definition = Mockery::mock();
+
+		$this->container_builder
+			->expects( 'getDefinition' )
+			->once()
+			->with( Schema_Blocks::class )
+			->andReturn( $schema_blocks_definition );
+
+		$expected_schema_blocks = [
```

Comment:
als je de get_templates in de compiler pass uitbesteedt aan een losse clas (wat meteen goed past bij separation of concern) dan kun je die class hier triviaal mocken.
klopt het dat deze deze test faalt zodra we een template toevoegen?

Reply:
That's a good idea 👍  How should I name such class? Something like: `template_repository` or `template_loader`

Ah and yes the test will fail when we add a template.